### DISCUSSION TITLE
Renovate beyond dependencies

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#15515] [Suggest an Idea]
We usually kickstart a new project using some base project. The new project then gets its own Git repository and has no relation whatsoever to the base project anymore. However, soon the base project gets updates, improvements, migrations, etc. These currently need to be cherry-picked manually (and sometimes painfully) to bring the derived project up to date. The longer this is avoided, the more changes pile up and become an issue or blocker in case of updates.

So far we do not have a good solution for this aside from thoroughly documenting the manual steps which need to be done in each project. Then someone :tm: has to do them.

Now I wonder if we could use Renovate to not only suggest dependency updates for projects but also updates to the project itself.

I am aware of the [`regex`](https://docs.renovatebot.com/modules/manager/regex/) manager which can be used apply custom version updates to custom files using one of the existing data sources. However, what I have in mind here goes way beyond that. It would mean knowledge of various file formats and how to modify them. Also at some points CLI commands may need to be involved to perform updates. Some adjustments can be done with CLI commands alone.

I am aware of tools like [multi-gitter](https://github.com/lindell/multi-gitter) and [git-xargs](https://blog.gruntwork.io/introducing-git-xargs-an-open-source-tool-to-update-multiple-github-repos-753f9f3675ec). Still I wonder if this would fit the idea of Renovate or if this is completely out of scope.

I am also open for suggestions if there are existing tools/solutions which do this.

One of the great things about Renovate is that its update PRs are idempotent; they are created if necessary and automatically updated (e.g. rebased) if necessary. Also they don't show up again once merged/closed. This obviously would be nice to have with a different tool.

**Comment by rarkins** [comment]
Here is an old idea, seems similar? https://github.com/renovatebot/renovate/issues/4759

**Reply by mbrodala** [reply]
Yes and no. I was thinking beyond file syncing. So something like a `patch` manager would be more in line with what I have in mind. Only that the actual `patch` would not be flexible enough and often fail to apply. ;-)
